server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -789,6 +791,63 @@ func extractOptionsFromStartedBuildEvent(event *build_event_stream.BuildEvent) (
 	return "", nil
 }
 
+func getCursesLinesFromStartedBuildEvent(event *build_event_stream.BuildEvent) int {
+	// the number of lines curses can overwrite is 2 + the ui_actions shown:
+	// 1 for the progress tracker, 1 for each action, and 1 blank line
+	options, err := extractOptionsFromStartedBuildEvent(event)
+	if err != nil {
+		log.Errorf("Could not extract options for ui_actions_show, defaulting to 8: %v", err)
+		return 10
+	}
+	optionsList, err := shlex.Split(options)
+	if err != nil {
+		log.Errorf("Could not shlex split options for ui_actions_show, defaulting to 8: %v", err)
+		return 10
+	}
+	actionsShownValues := getValuesForOptionNameFromOptionsList(optionsList, "ui_actions_shown")
+	cursesValues := getValuesForOptionNameFromOptionsList(optionsList, "ui_actions_shown")
```

Comment:
I think it would be cleaner if you split the logic to return the `ui_actions_shown` value into a separate function and then have this function add +2 to the value returned from this new function (and return 0 to mean that no lines should be buffered)

Reply:
I can do that.